packages/go_router/lib/src/go_router_delegate.dart

```diff
@@ -338,11 +338,16 @@ class GoRouterDelegate extends RouterDelegate<Uri>
         // get stack of route matches
         matches = _getLocRouteMatches(loc, extra: extra);
 
-        var params = <String, String>{};
+        // merge new params to keep params from previously matched paths, e.g.
+        // /family/:fid/person/:pid provides fid and pid to person/:pid
+        var previouslyMatchedParams = <String, String>{};
         for (final match in matches) {
-          // merge new params to keep params from previously matched paths, e.g.
-          // /family/:fid/person/:pid provides fid and pid to person/:pid
-          params = {...params, ...match.decodedParams};
+          assert(
+              !previouslyMatchedParams.keys
+                  .any(match.encodedParams.containsKey),
+              'Duplicated parameter names');
```

Comment:
nit: add a comma after `Duplicated parameter names` for formatting.
```
      assert(
        !previouslyMatchedParams.keys.any(match.encodedParams.containsKey),
        'Duplicated parameter names',
      );
```

Reply:
Done!